### PR TITLE
[JENKINS-41977] - Explicitly set the charset in HudsonAuthenticationEntryPoint

### PR DESCRIPTION
**Pull request by tkizm1**
Login page is missing charset.

### CODE REVIEW

**Comment by daniel-beck** [comment]
What's the issue this fixes?

**Comment by tkisme** [comment]
see [this](https://issues.jenkins-ci.org/browse/JENKINS-41977)

**Review by oleg-nenashev — APPROVED** [review]
The fix looks reasonable to me. 👍

File: core/src/main/java/hudson/security/HudsonAuthenticationEntryPoint.java [hunk 1]
```diff
@@ -102,6 +102,7 @@ public void commence(ServletRequest request, ServletResponse response, Authentic
             }
             out.printf(
                 "<html><head>" +
+                "<meta http-equiv='Content-Type' content='text/html;charset=utf-8'>" +
                 "<meta http-equiv='refresh' content='1;url=%1$s'/>" +
```
Comment (daniel-beck) [inline core/src/main/java/hudson/security/HudsonAuthenticationEntryPoint.java]: This does not seem to do what it claims to in the issue, as it sets the encoding for the HTTP redirect page, not the `/login` URL. Has this change been tested at all? What am I missing?

**Review by daniel-beck — CHANGES_REQUESTED** [review]
I have questions.

File: core/src/main/java/hudson/security/HudsonAuthenticationEntryPoint.java (hunk 1 above)
Comment (tkisme) [inline core/src/main/java/hudson/security/HudsonAuthenticationEntryPoint.java]: This error always come with https ,without https everything just fine,I will change all charset to see what happens.

Comment (tkisme) [inline core/src/main/java/hudson/security/HudsonAuthenticationEntryPoint.java]: This error always come with https without https everything just fine,I will change all charset to see what happens.

Comment (tkisme) [inline core/src/main/java/hudson/security/HudsonAuthenticationEntryPoint.java]: This error always come with https without https everything just fine,I will change all charset to see what happens.

Comment (tkisme) [inline core/src/main/java/hudson/security/HudsonAuthenticationEntryPoint.java]: now I found this error only occurs when define gzip compression.I close it.

Comment (tkisme) [inline core/src/main/java/hudson/security/HudsonAuthenticationEntryPoint.java]: If someone came with same error,you can see here
https://github.com/containous/traefik/issues/1060

**Pull request closed without merging**